ProjFS.Mac/PrjFSLib/PrjFSLib.cpp

```diff
@@ -866,6 +902,47 @@ static PrjFS_Result HandleFileNotification(
         nullptr /* destinationRelativePath */);
 }
 
+static void FindNewFoldersInRootAndNotifyProvider(const MessageHeader* request, const char* path)
+{
+    // Walk up the directory tree and notify the provider about any directories
+    // not flagged as being in the root
+    stack<pair<string /*relative path*/, string /*full path*/>> newFolderPaths;
+    string parentPath(path);
+    size_t lastDirSeparator = parentPath.find_last_of('/');
+    while (lastDirSeparator != string::npos && lastDirSeparator > 0)
+    {
+        parentPath = parentPath.substr(0, lastDirSeparator);
+        char parentFullPath[PrjFSMaxPath];
+        CombinePaths(s_virtualizationRootFullPath.c_str(), parentPath.c_str(), parentFullPath);
+        if (IsBitSetInFileFlags(parentFullPath, FileFlags_IsInVirtualizationRoot))
```

Comment:
Does it matter that we're setting the bit before we notified the provider? What happens if we lose power between here and there? Would it be more robust if we notify the provider first?